frontend/src/bundles/common/components/header/components/notifications/notification-component.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+    useCallback,
+    useHandleClickOutside,
+    useRef,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
+
+import { NotificationList } from './notification-list.js';
+import { NotificationIcon } from './notifications-header.js';
+
+const NotificationComponent = (): JSX.Element => {
+    const [notifications, setNotifications] = useState([
```

Comment:
maybe we can move this state to the store and add necessary actions as well at this point?